src/coord/src/catalog/storage.rs

```diff
@@ -681,25 +696,26 @@ impl<'a, S: Append> Transaction<'a, S> {
     /// Panics if any introspection source id is not a system id
     pub fn insert_compute_instance(
         &mut self,
+        id: ComputeInstanceId,
         cluster_name: &str,
         config: &Option<ComputeInstanceIntrospectionConfig>,
         introspection_sources: &Vec<(&'static BuiltinLog, GlobalId)>,
-    ) -> Result<ComputeInstanceId, Error> {
+    ) -> Result<(), Error> {
         let config = serde_json::to_string(config)
             .map_err(|err| Error::from(StashError::from(err.to_string())))?;
-        let id = match self.compute_instances.insert(
-            |id| ComputeInstanceKey { id: id.unwrap() },
+        match self.compute_instances.insert(
```

Comment:
This should now be a `if let Err(err) = ...`

Reply:
Fixed